### DISCUSSION TITLE
How to compile a Kernel via a custom IDE

### DISCUSSION THREAD

**Discussion by RetroAndDev** [CosmosOS/Cosmos#1759] [Q&A]
Hello, wanting to make a custom IDE to create and compile apps for my system but I don't know how to use the Cosmos compiler.
What are the methods used during the compilation? And which sources are used?
I use the laset dev kit
Thank you for your help

**Comment by quajak** [comment]
Hi, I moved this over to Discussions since its more fitting to discuss it here.
You should be able to run the Cosmos compiler using `dotnet run` if the project is set up correctly. 
The code in cosmos which is triggered by dotnet run is in Cosmos.Build.Tasks with the msbuild directions in `/build/Cosmos.Build.targets`.

**Reply by RetroAndDev** [reply]
Ok, the file for the build os Lanch.cs ? Or a other file ?

**Reply by quajak** [reply]
Building takes a series of steps, so there isnt a single file. You have to reproduce all the steps in the Cosmos.Build.Targets.

**Reply by RetroAndDev** [reply]
Ok, if I want to get the order of the methods, I can use the VS extension code?

**Reply by quajak** [reply]
The order is also set in `Cosmos.Build.Targets`

**Comment by ilobilo** [comment]
How can I install without visual studio?

**Reply by quajak** [reply]
if you just stop the installation where it tries to install the vs extensions, i think you should be fine if you get the nuget packages working. You won't be able to debug but should be able to build an iso and launch it manually.

**Reply by ilobilo** [reply]
Thanks

**Reply by ilobilo** [reply]
Also can I install it without visual studio on Linux?

**Reply by quajak** [reply]
No since Il2CPU requires windows specific libraries, there should be an issue somewhere tracking it.

**Reply by ilobilo** [reply]
Ok
